LanguageFeatures/Inline-classes/static_analysis_inline_class_A04_t01.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A compile-time error occurs if a type parameter of an inline
+/// class occurs in a non-covariant position in the representation type.
+///
+/// @description Checks that it is a compile-time error if a type parameter of
+/// an inline class occurs in a non-covariant position in the representation
+/// type
+/// @author sgrekhov22@gmail.com
+// SharedOptions=--enable-experiment=inline-class
+
+inline class V<T extends num> {
+  final void Function(T) f;
+//                    ^
+// [analyzer] unspecified
+// [cfe] unspecified
```

Comment:
Yes, thanks!

Reply:
@eernstg great! Thank you! We do have a natural location for such tests, but I don't see any tests for the statement below there. In the Dart Language spec we have:

> **10.10 Superinterfaces** 
> ...
> When a generic class C declares a type parameter X, it is a compile-time
> error if X occurs in a non-covariant position in a type which specifies a superinterface of C. _For example, the class can not have List<void Function(X)> in its extends or implements clause._
> 

And no any co19 for it. Looks like that this statement were added relatively recently. Created https://github.com/dart-lang/co19/issues/2066 for this. 

BTW without `inline` keyword
```dart
class V<T extends num> {
  final void Function(T) f;
//                    ^
// [analyzer] unspecified
// [cfe] unspecified
  V(this.f);
}

main() {
  print(V);
}
```
Fails in both analyzer and CFE (no expected error). Is this an issue?